### PR TITLE
Add support for Istio protocol selection in service port names

### PR DESCRIPTION
**Pull request by royantman**
### Type of change

- Enhancement / new feature

### Description
TL;DR Adding "tcp-" prefix to 9092/9093/9094 port names in services.

This PR adds a support for Istio protocol selection in service port names.
The change mainly resolves an issue with Istio creating a wildcard listener in Envoy (e.g 0.0.0.0:9093) and thus forces a single destination rule for all services using this port and also causes Mixer metrics to appear with wrong names when having multiple Strimzi clusters.
Also this can cause collisions with other services that also use the same port (9093) (e.g Prometheus alert manager)

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [X ] Make sure all tests pass
- [X ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [X ] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by royantman** [comment]
@scholzj afaik the only internal resource that has a relationship with the name of the port are Endpoints which needs a port name to match a service name.
K8s is creating and managing endpoints automagically in the case of selectors in Service (like Strimzi). If one is to define an endpoint manually it would be for external resource Service (e.g. an RDS or some other out-of-k8s-resource.

@tombentley https://istio.io/docs/ops/traffic-management/protocol-selection/
When using a [Sidecar](https://istio.io/docs/reference/config/networking/sidecar/) for example in order to avoid cluster-wide port conflicts, and having more than one headless service (2 clusters deployed), the resulting Envoy listener would be on 0.0.0.0_9093.

```apiVersion: networking.istio.io/v1alpha3
kind: Sidecar
metadata:
  name: test-client
spec:
  workloadSelector:
    labels:
      app: test-client
  egress:
  - hosts:
    - "alice-system/default-kafka-brokers.alice-system.svc.cluster.local"
    - "alice-system/other-kafka-brokers.alice-system.svc.cluster.local"
    - "istio-system/*"
```

With all that said... the dream change in Strimzi would be addressing #1506 
As you can see, no `port` is provided under `egress` in order for the result would be a listener per IP under the headless service.

This results in the following example of listeners in Envoy:
```
100.96.5.36_9093::100.96.5.36:9093
100.96.11.22_9091::100.96.11.22:9091
100.96.10.35_9093::100.96.10.35:9093
100.96.11.22_9093::100.96.11.22:9093
100.96.5.36_9091::100.96.5.36:9091
100.96.10.35_9091::100.96.10.35:9091
100.96.10.36_9093::100.96.10.36:9093
100.96.10.36_9091::100.96.10.36:9091
```

And there is no generic `0.0.0.0_9093` routing so all traffic is identified per-broker and thus all the metrics are also per-broker.

As per the docs (and source code) about `port` under `egress` field:
```
The port associated with the listener. If using Unix domain socket, 
use 0 as the port number, with a valid protocol. The port if specified, 
will be used as the default destination port associated with the imported hosts. 
If the port is omitted, Istio will infer the listener ports based on the imported hosts. 
Note that when multiple egress listeners are specified, where one or more listeners 
have specific ports while others have no port, the hosts exposed on a listener port 
will be based on the listener with the most specific port.
```

Just making sure it is clear. All this is relevant to the client to broker routing/metrics. We are not talking here about running Strimzi behind Envoy.

With all that said, the dream change in Strimzi would be addressing #1506 
and allow the users to avoid any port collision in the future, since it will be their own fault :)

**Comment by scholzj** [comment]
> @scholzj afaik the only internal resource that has a relationship with the name of the port are Endpoints which needs a port name to match a service name.

I meant more if someone of our users is using the names for some logic. That is why I think it should be explicitely mentioned int he CHANGELOG.

> With all that said, the dream change in Strimzi would be addressing #1506
> and allow the users to avoid any port collision in the future, since it will be their own fault :)

I think we might get to that sooner or later as well. There have been several issues / discussions related to that recently. So hopefully we get to something as well there.

**Comment by royantman** [comment]
Super

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,9 @@
 * Garbage Collection (GC) logging disabled by default
 * Providing PKCS12 truststore and password in the cluster and clients CA certificates Secrets
 * Providing PKCS12 keystore and password in the TLS based KafkaUser related Secret
-
+* Add support for Istio protocol selection in service port names  
+Note: Strimzi is essentially addind a `tcp-` prefix to the port names in Kafka services and headless services.
```
Comment (tombentley) [inline CHANGELOG.md]:
```suggestion
Note: Strimzi is essentially adding a `tcp-` prefix to the port names in Kafka services and headless services.  
```

**Comment by royantman** [comment]
Any chance this is getting into 0.15 and not 0.16 ? 
We rather not run with a fork in our environment.

**Comment by scholzj** [comment]
I will cherry-pick it if we fo another RC. In the mean time I will merge it to master - I hope we do 0.16.0 before or during December.

**Comment by scholzj** [comment]
(And thanks for the PR!)
